api/src/services/comments.ts

```diff
@@ -0,0 +1,466 @@
+		const commentsQuery = this.serviceOrigin === 'comments' ? query : this.generateQuery('comments', query);
+		const activityResult = await this.activityService.readByQuery(activityQuery, opts);
+		const commentsResult = await super.readByQuery(commentsQuery, opts);
+
+		if (query.aggregate) {
+			// Merging the first result only as the app does not utilise group
+			return [
+				mergeWith({}, activityResult[0], commentsResult[0], (a: any, b: any) => {
+					const numA = Number(a);
+					const numB = Number(b);
+
+					if (!isNaN(numA) && !isNaN(numB)) {
+						return numA + numB;
+					}
+
+					return;
+				}),
+			];
+		} else if (query.sort) {
+			return this.sortLegacyResults([...activityResult, ...commentsResult], query.sort);
+		} else {
+			return [...activityResult, ...commentsResult];
+		}
+	}
+
+	override async readMany(keys: PrimaryKey[], query?: Query, opts?: QueryOptions): Promise<Item[]> {
+		const commentsKeys = [];
+		const activityKeys = [];
+
+		for (const key of keys) {
+			if (isNaN(Number(key))) {
+				commentsKeys.push(key);
+			} else {
+				activityKeys.push(key);
+			}
+		}
+
+		const activityQuery = this.serviceOrigin === 'activity' ? query : this.generateQuery('activity', query || {});
+		const commentsQuery = this.serviceOrigin === 'comments' ? query : this.generateQuery('comments', query || {});
+		const activityResult = await this.activityService.readMany(activityKeys, activityQuery, opts);
+		const commentsResult = await super.readMany(commentsKeys, commentsQuery, opts);
+
+		if (query?.sort) {
+			return this.sortLegacyResults([...activityResult, ...commentsResult], query.sort);
+		} else {
+			return [...activityResult, ...commentsResult];
+		}
+	}
+
+	override async createOne(data: Partial<Comment>, opts?: MutationOptions): Promise<PrimaryKey> {
+		if (!data['comment']) {
+			throw new InvalidPayloadError({ reason: `"comment" is required` });
+		}
+
+		if (!data['collection']) {
+			throw new InvalidPayloadError({ reason: `"collection" is required` });
+		}
+
+		if (!data['item']) {
+			throw new InvalidPayloadError({ reason: `"item" is required` });
+		}
+
+		if (this.accountability) {
+			await validateAccess(
+				{
+					accountability: this.accountability,
+					action: 'read',
+					collection: data['collection'],
+					primaryKeys: [data['item']],
+				},
+				{
+					schema: this.schema,
+					knex: this.knex,
+				},
+			);
+		}
+
+		const usersRegExp = new RegExp(/@[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}/gi);
+
+		const mentions = uniq(data['comment'].match(usersRegExp) ?? []);
+
+		const sender = await this.usersService.readOne(this.accountability!.user!, {
+			fields: ['id', 'first_name', 'last_name', 'email'],
+		});
+
+		for (const mention of mentions) {
+			const userID = mention.substring(1);
+
+			const user = await this.usersService.readOne(userID, {
+				fields: ['id', 'first_name', 'last_name', 'email', 'role.id', 'role.admin_access', 'role.app_access'],
```

Comment:
Looks like this slipped the review and is causing this issue: https://github.com/directus/directus/issues/23970

`role.admin_access` does no longer and should be replaced by `const { admin, app } = await fetchGlobalAccess(accountability, database);`

Reply:
Resolved via https://github.com/directus/directus/pull/23972